### ISSUE TITLE
accuracy, recall, precision and f1-score are equal

### ISSUE THREAD

**Issue by Lucienxhh** [Lightning-AI/metrics#1111]
## 🐛 Bug

when i evaluate my model following the demo provided [here](https://torchmetrics.readthedocs.io/en/stable/pages/quickstart.html), i found the results were strange that accuracy, recall, precision and f1-score are equal.

### To Reproduce

```py
import torch
import torchmetrics

torch.manual_seed(0)
batches = 10
test_accuracy = torchmetrics.Accuracy()
test_recall = torchmetrics.Recall()
test_precision = torchmetrics.Precision()
test_f1 = torchmetrics.F1Score()

for i in range(batches):
    preds = torch.randn(10,5).softmax(dim=-1)
    target = torch.randint(5, (10,))
    test_accuracy(preds, target)
    test_recall(preds, target)
    test_precision(preds, target)
    test_f1(preds, target)
    
acc = test_accuracy.compute()
recall = test_recall.compute()
precision = test_precision.compute()
f1 = test_f1.compute()

print(acc, recall, precision, f1)
```

exactly, the accuracy, recall, precision and f1-score should be different because of the randomly created `preds`
### Environment

ubuntu==18.04, pip==21.2.2, torchmetrics==0.8.2, torch=1.9.0+cu111

**Comment by Borda** [comment]
seems to be the same issue, closing in favor of #746

**Comment by Lucienxhh** [comment]
@Borda thanks for your advice.
i follow your suggestion that set the average as 'macro', but it doesn't work
here is the code that compares `torchmetrics` and `sklearn.metrics`
```
import torch
import torchmetrics
from sklearn import metrics
torch.manual_seed(4)
batches = 10
num_classes = 2

# torchmetric init
average = 'macro' if num_classes == 2 else 'micro'
torchmetrics_accuracy = torchmetrics.Accuracy()
torchmetrics_recall = torchmetrics.Recall(average=average, num_classes=num_classes)
torchmetrics_precision = torchmetrics.Precision(average=average, num_classes=num_classes)
torchmetrics_f1 = torchmetrics.F1Score(average=average, num_classes=num_classes)

pred = []
real = []
for i in range(batches):
    preds = torch.randn(100, num_classes).softmax(dim=-1)
    target = torch.randint(2, (100,))
    
    # torchmetrics
    torchmetrics_accuracy(preds, target)
    torchmetrics_precision(preds, target)
    torchmetrics_recall(preds, target)
    torchmetrics_f1(preds, target)
    
    # sklearn.metrics
    preds_ = preds.cpu().numpy()
    idx = preds_.argmax(axis=1)
    pred.extend(idx.tolist())
    real.extend(target.cpu().numpy().tolist())


# calculation of torchmetrics
acc_1 = torchmetrics_accuracy.compute()
precision_1 = torchmetrics_precision.compute()
recall_1 = torchmetrics_recall.compute()
f1_1 = torchmetrics_f1.compute()
print(acc_1, recall_1, precision_1, f1_1)

# calculation of torchmetrics
average = 'binary' if num_classes == 2 else 'micro'
acc_2 = metrics.accuracy_score(real, pred)
precision_2 = metrics.precision_score(real, pred, average=average)
recall_2 = metrics.recall_score(real, pred, average=average)
f1_2 = metrics.f1_score(real, pred, average=average)
print(acc_2, recall_2, precision_2, f1_2)
```

i have read the doc of torchmetrics and sklearn.metrics, and found that torchmetrics didn't have `binary` option for `average`.
details are here: [sklearn.metrics](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score)  and [torchmetrics](https://torchmetrics.readthedocs.io/en/stable/classification/f1_score.html)
